Civi/Core/Container.php

```diff
@@ -159,14 +159,17 @@ public function createContainer() {
       'groups' => 'contact groups',
     ];
     foreach ($basicCaches as $cacheSvc => $cacheGrp) {
+      $definitionParams = [
+        'name' => $cacheGrp,
+        'type' => ['*memory*', 'SqlGroup', 'ArrayCache'],
+      ];
+      $fastArrayCaches = ['groups'];
```

Comment:
Can we add a code comment saying that caches that are hit frequently & which have a low risk of concurrency issues (ie. session is riskier) should be defined as fastArrayCaches so that they use a static php array within a given process. This can speed up longer running jobs, in particular

Reply:
i added a comment but may not fully cover what you want

Reply:
ddd decorator

Reply:
goes with behabiour